modules/shadertools/test/modules-webgl1/fp64/fp64-test-utils-transform.ts

```diff
@@ -60,23 +60,22 @@ function setupFloatData({limit, op, testCases}) {
 function setupFloatTest(device: Device, {glslFunc, binary = false, limit = 256, op, testCases}) {
   const {a, b, expected, a_fp64, b_fp64, expected_fp64} = setupFloatData({limit, op, testCases});
   const vs = binary ? getBinaryShader(glslFunc) : getUnaryShader(glslFunc);
+  const bufferA = device.createBuffer({data: a_fp64});
+  const bufferB = device.createBuffer({data: b_fp64});
+  const bufferResult = device.createBuffer({byteLength: a_fp64.byteLength});
   const transform = new Transform(device, {
-    sourceBuffers: {
-      a: device.createBuffer({data: a_fp64}),
-      b: device.createBuffer({data: b_fp64})
-    },
     vs,
     modules: [fp64],
-    feedbackMap: {
-      a: 'result'
-    },
+    attributes: {a: bufferA, b: bufferB},
+    bufferLayout: [{name: 'a', format: 'float32x2'}, {name: 'b', format: 'float32x2'}],
+    feedbackBuffers: {result: bufferResult},
     varyings: ['result'],
-    elementCount: testCases.length
+    vertexCount: testCases.length
```

Comment:
NOTE! I'm using three buffers for these tests. Previously we used two, reading and writing to buffer `a`. I couldn't figure out how to set that up with the current Transform+TransformFeedback+Buffer APIs, and ran into these errors while trying:

```
GL_INVALID_OPERATION: A transform feedback buffer that would be written to is also
bound to a non-transform-feedback target, which would cause undefined behavior.
```

We shouldn't need three, hopefully that's something we can fix with the upcoming Transform API.

Reply:
I am probably not adding anything you haven't considered, but this says the buffer can't be bound to two places at the the time of the draw call. I.e. it can't be in the vertexArray and the transformFeedback at the same time. You'll need to keep switching the "input" buffers in both vertexArray, and the output buffers in the transformFeedback, and for good measure make sure we still bind()/unbind() the transform feedback buffers before/after use.